flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -879,4 +879,117 @@
 	 * @return An OverWindowedTable to specify the aggregations.
 	 */
 	OverWindowedTable window(OverWindow... overWindows);
+
+	/**
+	 * Adds additional columns. Similar to a SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. It will throw an exception
+	 * if the added fields already exist.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns("a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(String fields);
+
+	/**
+	 * Adds additional columns. Similar to a SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. It will throw an exception
+	 * if the added fields already exist.
+	 *
+	 * <p>Scala Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns('a + 1 as 'a1, concat('b, "sunny") as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(Expression... fields);
+
+	/**
+	 * Adds additional columns. Similar to a SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. Existing fields will be
+     * replaced if add columns name is the same as the existing column name. Moreover, if the added
+     * fields have duplicate field name, then the last one is used.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addOrReplaceColumns("a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addOrReplaceColumns(String fields);
+
+	/**
+	 * Adds additional columns. Similar to a SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. Existing fields will be
+	 * replaced. If the added fields have duplicate field name, then the last one is used.
```

Comment:
```suggestion
	 * replaced. If the added fields have duplicated field name, then the last one is used.
```

Reply:
Why we should do this change? `duplicate` -> `duplicated`?  I using `translate.google.cn`, It suggests using `duplicate`.  Can you double check this?